pkg/agent/util/ndp/ndp.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ndp
+
+import (
+	"fmt"
+	"io"
+	"net"
+
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv6"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// The assumed NDP option length (in units of 8 bytes) for fixed length options.
+	llaOptLen = 1
+
+	// Length of a link-layer address for Ethernet networks.
+	ethAddrLen = 6
+
+	// Type values for each type of valid Option.
+	optSourceLLA = 1
+	optTargetLLA = 2
+
+	// Minimum byte length values for each type of valid Message.
+	naLen = 20
+
+	// HopLimit is the expected IPv6 hop limit for all NDP messages.
+	HopLimit = 255
+)
+
+type INDPAdvertiser interface {
```

Comment:
Not sure if it's overkill to have an interface, a few structs, a function to do this. For IPv4, it's just a function. Does the struct provide some cache or save some repeated work when sending multiple NDP packets? If not, could we start from a function?

Reply:
ipv6.PacketConn can be reused when sending multiple NDP packets.